engine/access/rpc/engine.go

```diff
@@ -51,7 +54,11 @@ type Config struct {
 // An unsecured GRPC server (default port 9000), a secure GRPC server (default port 9001) and an HTTP Web proxy (default
 // port 8000) are brought up.
 type Engine struct {
+	component.Component
+	*events.FinalizationActor
+
 	unit               *engine.Unit
```

Comment:
```suggestion
```

Reply:
but it requires removing `SubmitLocal` which is either way a deprecated pattern

Reply:
There were some odd dependencies here, which I've cleaned up in https://github.com/onflow/flow-go/pull/4313/commits/141c474b0ef80556a7fd5efad556bd29ad7567ee and https://github.com/onflow/flow-go/pull/4313/commits/cdc79ab9a8de91d359de5e13e7adc286f269159d.

The `ingestion` engine depended on the `rpc` engine, but only used `SubmitLocal` and only to forward block finalization events from hotstuff 🤔. I changed this to remove the dependency so the `rpc` engine just subscribes to events directly.

Reply:
I was looking at this recently as well. I think it was added to implement tx submission retries before the finalization distributor was created.